google-cloud-dialogflow/src/test/java/com/google/cloud/dialogflow/v2/it/ITSystemTest.java

```diff
@@ -0,0 +1,323 @@
+            .setParent(PROJECT_AGENT_NAME.toString())
+            .setEntityType(createEntityType)
+            .build();
+    entityType = entityTypesClient.createEntityType(entityTypeRequest);
+    entityName =
+        entityType.getName().substring(entityType.getName().lastIndexOf("/")).replace("/", "");
+    entityTypeName = EntityTypeName.of(PROJECT_ID, entityName);
+
+    /* create intents */
+    intentsClient = IntentsClient.create();
+    Intent createIntent =
+        Intent.newBuilder()
+            .setDisplayName(INTENT_NAME)
+            .addEvents(EVENT_NAME)
+            .setAction(ACTION_NAME)
+            .build();
+    CreateIntentRequest intentRequest =
+        CreateIntentRequest.newBuilder()
+            .setParent(PROJECT_AGENT_NAME.toString())
+            .setIntent(createIntent)
+            .build();
+    intent = intentsClient.createIntent(intentRequest);
+    intentName = intent.getName().substring(intent.getName().lastIndexOf("/")).replace("/", "");
+    intent_name = IntentName.of(PROJECT_ID, intentName);
+
+    /* create session */
+    sessionsClient = SessionsClient.create();
+
+    /* create context */
+    contextsClient = ContextsClient.create();
+    Context createContext =
+        Context.newBuilder()
+            .setName(CONTEXT_NAME.toString())
+            .setLifespanCount(LIFE_SPAN_COUNT)
+            .build();
+    CreateContextRequest request =
+        CreateContextRequest.newBuilder()
+            .setParent(SESSION_NAME.toString())
+            .setContext(createContext)
+            .build();
+    context = contextsClient.createContext(request);
+  }
+
+  @AfterClass
+  public static void afterClassTest() {
+
+    /* delete context */
+    DeleteContextRequest deleteContextRequest =
+        DeleteContextRequest.newBuilder().setName(CONTEXT_NAME.toString()).build();
+    contextsClient.deleteContext(deleteContextRequest);
+    contextsClient.close();
+
+    /* delete session */
+    sessionsClient.close();
+
+    /* delete intents */
+    DeleteIntentRequest deleteIntentRequest =
+        DeleteIntentRequest.newBuilder().setName(intent_name.toString()).build();
+    intentsClient.deleteIntent(deleteIntentRequest);
+    intentsClient.close();
+
+    /* delete entity */
+    DeleteEntityTypeRequest deleteEntityTypeRequest =
+        DeleteEntityTypeRequest.newBuilder().setName(entityTypeName.toString()).build();
+    entityTypesClient.deleteEntityType(deleteEntityTypeRequest);
+    entityTypesClient.close();
+
+    /* delete agent */
+    DeleteAgentRequest deleteAgentRequest =
+        DeleteAgentRequest.newBuilder().setParent(PROJECT_NAME.toString()).build();
+    agentsClient.deleteAgent(deleteAgentRequest);
+    agentsClient.close();
+  }
+
+  @Test
+  public void searchAgentsTest() {
+    SearchAgentsRequest request =
+        SearchAgentsRequest.newBuilder().setParent(PROJECT_NAME.toString()).build();
+    for (Agent actual : agentsClient.searchAgents(request).iterateAll()) {
```

Comment:
rename to `actualAgent` so that anyone can understand what actual instance contains.

Reply:
Done.